packages/space-opera/src/components/materials_panel/materials_panel.ts

```diff
@@ -125,12 +137,65 @@ export class MaterialPanel extends ConnectedLitElement {
     this.safeUrlIds = safeUrlIds;
   }
 
+  /* Interpolate from red to original color as n approaches maxN */
+  getInterpolatedColor(original: RGBA, n: number, maxN: number): RGBA {
+    const red: RGBA = [1, 0, 0, 1];
+    const percentRed = (maxN - n) / maxN;
+    const percentOriginal = 1 - percentRed;
+    return [
+      (percentRed * red[0]) + (percentOriginal * original[0]),
+      (percentRed * red[1]) + (percentOriginal * original[1]),
+      (percentRed * red[2]) + (percentOriginal * original[2]),
+      original[3],
+    ];
+  }
+
+  interpolateMaterial() {
+    const index = this.selectedMaterialId!;
+
+    // TODO: Use emissive when changing said property won't reload the model.
+    // (switch dispatches as well when moving to emissive).
+    const originalBaseFactor = this.materials[index].baseColorFactor;
+
+    // The interval (in milliseconds) on how often to execute the code.
+    const intervalSpeed = 10;  // setInterval() has a minimum val of 10
+    let n = 0;
+    const maxN = 165;  // total number of executions
+
+    // Interpolate from red to the original color.
+    // Total time to complete depends on the intervalSpeed and maxN.
+    const interval = setInterval(() => {
+      n += 1;
```

Comment:
Let's put this in terms of time instead of arbitrary integers so that it can be called with arbitrary time deltas.